aten/src/ATen/native/RangeUtils.h

```diff
@@ -0,0 +1,45 @@
+#include <ATen/AccumulateType.h>
+#include <c10/core/Scalar.h>
+#include <limits>
+
+namespace at {
+
+namespace native {
+
+template <typename scalar_t>
+int64_t compute_arange_size(const Scalar& start, const Scalar& end, const Scalar& step) {
+  using accscalar_t = at::acc_type<scalar_t, false>;
+  auto xstart = start.to<accscalar_t>();
+  auto xend = end.to<accscalar_t>();
+  auto xstep = step.to<accscalar_t>();
+
+  TORCH_CHECK(xstep > 0 || xstep < 0, "step must be nonzero");
+  TORCH_CHECK(std::isfinite(static_cast<double>(xstart)) &&
```

Comment:
nit: you could skip these casts entirely if the types are integral? Or are we doing this for safety because int8/int4 types might have poor STL support

Reply:
I kept the code intact because I wasn't sure why it was there and didn't want to add any risks of regression, but `std::isfinite(static_cast<double>(xstart))` should be equivalent to `std::isfinite(xstart)`. Maybe int4 would be an edge case here?